AssertionError: If the instructor hasn't been retrieved yet there is some problem in adding of instructor

Received several of these yesterday and today, from the live server
v 6.11.0
----
Error Message: teammates.common.util.Logger severe: Exception occured while performing http://teammatesv4.appspot.com/worker/instructorCourseJoinEmail|||
java.lang.AssertionError: If the instructor hasn't been retrieved yet there is some problem in adding of instructor
teammates.common.util.retry.MaximumRetriesExceededException: getInstructor failed after maximum retries
at teammates.common.util.retry.RetryManager.doRetry(RetryManager.java:111)
at teammates.common.util.retry.RetryManager.runUntilNotNull(RetryManager.java:83)
at teammates.ui.automated.InstructorCourseJoinEmailWorkerAction.execute(InstructorCourseJoinEmailWorkerAction.java:54)
at teammates.ui.automated.AutomatedServlet.doPost(AutomatedServlet.java:46)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)

at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:848)
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1772)
at com.googlecode.objectify.ObjectifyFilter.doFilter(ObjectifyFilter.java:48)
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1759)
at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1759)
at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:582)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
at com.google.apphosting.runtime.jetty9.ParseBlobUploadHandler.handle(ParseBlobUploadHandler.java:120)
at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1182)
at com.google.apphosting.runtime.jetty9.AppEngineWebAppContext.doHandle(AppEngineWebAppContext.java:171)
at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512)
at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
at com.google.apphosting.runtime.jetty9.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:297)
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
at org.eclipse.jetty.server.Server.handle(Server.java:539)
at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:333)
at com.google.apphosting.runtime.jetty9.RpcConnection.handle(RpcConnection.java:202)
at com.google.apphosting.runtime.jetty9.RpcConnector.serviceRequest(RpcConnector.java:81)
at com.google.apphosting.runtime.jetty9.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:123)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchServletRequest(JavaRuntime.java:692)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchRequest(JavaRuntime.java:654)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:624)
at com.google.apphosting.runtime.JavaRuntime$NullSandboxRequestRunnable.run(JavaRuntime.java:818)
at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:274)
at java.lang.Thread.run(Thread.java:745)

at teammates.common.util.Assumption.fail(Assumption.java:61)
at teammates.ui.automated.InstructorCourseJoinEmailWorkerAction.execute(InstructorCourseJoinEmailWorkerAction.java:62)
at teammates.ui.automated.AutomatedServlet.doPost(AutomatedServlet.java:46)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:848)
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1772)
at com.googlecode.objectify.ObjectifyFilter.doFilter(ObjectifyFilter.java:48)
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1759)
at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1759)
at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:582)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
at com.google.apphosting.runtime.jetty9.ParseBlobUploadHandler.handle(ParseBlobUploadHandler.java:120)
at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1182)
at com.google.apphosting.runtime.jetty9.AppEngineWebAppContext.doHandle(AppEngineWebAppContext.java:171)
at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:512)
at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112)
at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
at com.google.apphosting.runtime.jetty9.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:297)
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
at org.eclipse.jetty.server.Server.handle(Server.java:539)
at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:333)
at com.google.apphosting.runtime.jetty9.RpcConnection.handle(RpcConnection.java:202)
at com.google.apphosting.runtime.jetty9.RpcConnector.serviceRequest(RpcConnector.java:81)
at com.google.apphosting.runtime.jetty9.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:123)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchServletRequest(JavaRuntime.java:692)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.dispatchRequest(JavaRuntime.java:654)
at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:624)
at com.google.apphosting.runtime.JavaRuntime$NullSandboxRequestRunnable.run(JavaRuntime.java:818)
at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:274)
at java.lang.Thread.run(Thread.java:745)

--- a/src/main/java/teammates/ui/automated/InstructorCourseJoinEmailWorkerAction.java
+++ b/src/main/java/teammates/ui/automated/InstructorCourseJoinEmailWorkerAction.java
@@ -7,9 +7,6 @@ import teammates.common.exception.TeammatesException;
 import teammates.common.util.Assumption;
 import teammates.common.util.Const.ParamsNames;
 import teammates.common.util.EmailWrapper;
-import teammates.common.util.retry.MaximumRetriesExceededException;
-import teammates.common.util.retry.RetryManager;
-import teammates.common.util.retry.RetryableTaskReturns;
 import teammates.logic.api.EmailGenerator;
 
 /**
@@ -47,21 +44,7 @@ public class InstructorCourseJoinEmailWorkerAction extends AutomatedAction {
         // saving the new instructor takes more time in GAE.
         // The instructor `id` can be constructed back using (instructorEmail%courseId)
         // because instructors' email cannot be changed before joining the course.
-        RetryManager rm = new RetryManager(4);
-        InstructorAttributes instructor = null;
-
-        try {
-            instructor = rm.runUntilNotNull(new RetryableTaskReturns<InstructorAttributes>("getInstructor") {
-                @Override
-                public InstructorAttributes run() {
-                    return logic.getInstructorById(courseId, instructorEmail);
-                }
-            });
-        } catch (MaximumRetriesExceededException e) {
-            // This Assumption fail ensures that the task is retried
-            Assumption.fail("If the instructor hasn't been retrieved yet there is some problem in adding of instructor"
-                    + TeammatesException.toStringWithStackTrace(e));
-        }
+        InstructorAttributes instructor = logic.getInstructorById(courseId, instructorEmail);
         Assumption.assertNotNull(instructor);
 
         EmailWrapper email = new EmailGenerator()

--- a/src/main/java/teammates/ui/automated/StudentCourseJoinEmailWorkerAction.java
+++ b/src/main/java/teammates/ui/automated/StudentCourseJoinEmailWorkerAction.java
@@ -6,9 +6,6 @@ import teammates.common.exception.TeammatesException;
 import teammates.common.util.Assumption;
 import teammates.common.util.Const.ParamsNames;
 import teammates.common.util.EmailWrapper;
-import teammates.common.util.retry.MaximumRetriesExceededException;
-import teammates.common.util.retry.RetryManager;
-import teammates.common.util.retry.RetryableTaskReturns;
 import teammates.logic.api.EmailGenerator;
 
 /**
@@ -39,20 +36,7 @@ public class StudentCourseJoinEmailWorkerAction extends AutomatedAction {
         CourseAttributes course = logic.getCourse(courseId);
         Assumption.assertNotNull(course);
 
-        RetryManager rm = new RetryManager(4);
-        StudentAttributes student = null;
-
-        try {
-            student = rm.runUntilNotNull(new RetryableTaskReturns<StudentAttributes>("getStudent") {
-                @Override
-                public StudentAttributes run() throws RuntimeException {
-                    return logic.getStudentForEmail(courseId, studentEmail);
-                }
-            });
-        } catch (MaximumRetriesExceededException e) {
-            Assumption.fail("If the student hasn't been retrieved yet there is some problem in adding of student"
-                    + TeammatesException.toStringWithStackTrace(e));
-        }
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmail);
         Assumption.assertNotNull(student);
 
         EmailWrapper email = isRejoin

--- a/src/main/java/teammates/ui/controller/InstructorCourseInstructorAddAction.java
+++ b/src/main/java/teammates/ui/controller/InstructorCourseInstructorAddAction.java
@@ -20,28 +20,23 @@ public class InstructorCourseInstructorAddAction extends InstructorCourseInstruc
 
         String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
         Assumption.assertPostParamNotNull(Const.ParamsNames.COURSE_ID, courseId);
-        String instructorName = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_NAME);
-        Assumption.assertPostParamNotNull(Const.ParamsNames.INSTRUCTOR_NAME, instructorName);
-        String instructorEmail = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_EMAIL);
-        Assumption.assertPostParamNotNull(Const.ParamsNames.INSTRUCTOR_EMAIL, instructorEmail);
 
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
         gateKeeper.verifyAccessible(
                 instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
 
-        InstructorAttributes instructorToAdd = extractCompleteInstructor(
-                courseId, instructorName, instructorEmail);
+        InstructorAttributes instructorToAdd = extractCompleteInstructor(courseId);
 
         /* Process adding the instructor and setup status to be shown to user and admin */
         try {
             logic.createInstructor(instructorToAdd);
             taskQueuer.scheduleCourseRegistrationInviteToInstructor(
-                    loggedInUser.googleId, instructorEmail, courseId);
+                    loggedInUser.googleId, instructorToAdd.email, instructorToAdd.courseId);
 
-            statusToUser.add(new StatusMessage(String.format(Const.StatusMessages.COURSE_INSTRUCTOR_ADDED,
-                                                             instructorName, instructorEmail),
-                                               StatusMessageColor.SUCCESS));
-            statusToAdmin = "New instructor (<span class=\"bold\"> " + instructorEmail + "</span>)"
+            statusToUser.add(new StatusMessage(
+                    String.format(Const.StatusMessages.COURSE_INSTRUCTOR_ADDED, instructorToAdd.name, instructorToAdd.email),
+                    StatusMessageColor.SUCCESS));
+            statusToAdmin = "New instructor (<span class=\"bold\"> " + instructorToAdd.email + "</span>)"
                     + " for Course <span class=\"bold\">[" + courseId + "]</span> created.<br>";
         } catch (EntityAlreadyExistsException e) {
             setStatusForException(e, Const.StatusMessages.COURSE_INSTRUCTOR_EXISTS);
@@ -56,16 +51,20 @@ public class InstructorCourseInstructorAddAction extends InstructorCourseInstruc
 
     /**
      * Creates a new instructor with all information filled in, using request parameters.
-     * This includes basic information as well as custom privileges (if applicable).
      *
-     * @param courseId        Id of the course the instructor is being added to.
-     * @param instructorName  Name of the instructor.
-     * @param instructorEmail Email of the instructor.
+     * <p>This includes basic information as well as custom privileges (if applicable).
+     *
+     * @param courseId Id of the course the instructor is being added to.
      * @return An instructor with all relevant info filled in.
      */
-    private InstructorAttributes extractCompleteInstructor(String courseId, String instructorName, String instructorEmail) {
+    private InstructorAttributes extractCompleteInstructor(String courseId) {
+        String instructorName = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_NAME);
+        Assumption.assertPostParamNotNull(Const.ParamsNames.INSTRUCTOR_NAME, instructorName);
+        String instructorEmail = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_EMAIL);
+        Assumption.assertPostParamNotNull(Const.ParamsNames.INSTRUCTOR_EMAIL, instructorEmail);
         String instructorRole = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_ROLE_NAME);
         Assumption.assertPostParamNotNull(Const.ParamsNames.INSTRUCTOR_ROLE_NAME, instructorRole);
+
         boolean isDisplayedToStudents = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_IS_DISPLAYED_TO_STUDENT) != null;
         String displayedName = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_DISPLAY_NAME);
         if (displayedName == null || displayedName.isEmpty()) {

--- a/src/test/java/teammates/test/cases/action/InstructorCourseInstructorAddActionTest.java
+++ b/src/test/java/teammates/test/cases/action/InstructorCourseInstructorAddActionTest.java
@@ -42,8 +42,8 @@ public class InstructorCourseInstructorAddActionTest extends BaseActionTest {
 
         String[] submissionParams = new String[] {
                 Const.ParamsNames.COURSE_ID, courseId,
-                Const.ParamsNames.INSTRUCTOR_NAME, newInstructorName,
-                Const.ParamsNames.INSTRUCTOR_EMAIL, newInstructorEmail,
+                Const.ParamsNames.INSTRUCTOR_NAME, newInstructorName + " ", // with some extra spaces
+                Const.ParamsNames.INSTRUCTOR_EMAIL, " " + newInstructorEmail + " ", // with some extra spaces
 
                 Const.ParamsNames.INSTRUCTOR_ROLE_NAME,
                 Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER,

